shaders/__usd/__UsdPreviewSurfaceParameters.osl

```diff
@@ -0,0 +1,196 @@
+//
+shader UsdPreviewSurfaceParameters
+(
+    // Inputs
+    color diffuseColor = color(0.18, 0.18, 0.18),
+    color emissiveColor = color(0.0, 0.0, 0.0),
+    int useSpecularWorkflow = 0,
+    color specularColor = color(0.0, 0.0, 0.0),
+    float metallic = 0.0,
+    float roughness = 0.5,
+    float clearcoat = 0.0,
+    float clearcoatRoughness = 0.01,
+    float opacity = 1.0,
+    float opacityThreshold = 0.0,
+    float ior = 1.5,
+    normal normalIn = normal(0.0, 0.0, 1.0),
+    float displacement = 0.0,
+    float occlusion = 1.0,
+
+    // Diffuse outputs
+    output float diffuseGainOut = 0.0
+        [[ string widget = "null" ]],
+    output color diffuseColorOut = color(0.0,0.0,0.0)
+        [[ string widget = "null" ]],
+
+    // Specular outputs
+    output color specularFaceColorOut = color(0.0,0.0,0.0)
+        [[ string widget = "null" ]],
+    output color specularEdgeColorOut = color(0.0,0.0,0.0)
+        [[ string widget = "null" ]],
+    output float specularRoughnessOut = 0.0
+        [[ string widget = "null" ]],
+    output color specularIorOut = color(0.0)
+        [[ string widget = "null" ]],
+
+    // Clearcoat outputs
+    output color clearcoatFaceColorOut = color(0.0,0.0,0.0)
+        [[ string widget = "null" ]],
+    output color clearcoatEdgeColorOut = color(0.0,0.0,0.0)
+        [[ string widget = "null" ]],
+    output float clearcoatRoughnessOut = 0.0
+        [[ string widget = "null" ]],
+
+    // Emissive outputs
+    output float glowGainOut = 0.0
+        [[ string widget = "null" ]],
+    output color glowColorOut = color(0.0,0.0,0.0)
+        [[ string widget = "null" ]],
+
+    // Normal outputs
+    output normal bumpNormalOut = normal(0.0,0.0,0.0)
+        [[ string widget = "null" ]],
+
+    // Opacity & Presence outputs
+    output float glassIorOut= 0.0
+        [[ string widget = "null" ]],
+    output float glassRoughnessOut = 0.0
+        [[ string widget = "null" ]],
+    output float refractionGainOut = 0.0
+        [[ string widget = "null" ]],
+    output float presenceOut = 1.0
+        [[ string widget = "null" ]],
+    
+    // Displacement outputs
+    output float dispAmountOut = 0.0
+        [[ string widget = "null" ]],
+    output float dispScalarOut = 0.0
+        [[ string widget = "null" ]],
+)
+{
+    // To achieve double-sided materials from a geometry-level property,
+    // we set a special user:hydra:doubleSided attribute during renderParam
+    // setup (see hdPrman/renderParam.cpp) and read that here. We also check 
+    // whether we are shading the front or back based on backfacing().
+    // 
+    // NB: We have to do this test in the shader because the *DoubleSided
+    // parameters of PxrSurface are NOT CONNECTABLE, so we cannot just switch
+    // them based on an output from this shader. We have to make the backface-
+    // shading decision here.
+    
+    int isDoubleSided = 0;
+    getattribute("user:hydra:doubleSided", isDoubleSided);
```

Comment:
We'll need to adapt this to be more Gaffer-specific, changing the `isDoubleSided` fallback to `1` to match Gaffer's default behaviour and renaming `user:hydra:doubleSided`. We'd also need to shuffle Gaffer's `doubleSided` attribute to that new name, either in a render adaptor or in the RenderMan backend itself, maybe @johnhaddon has some thoughts here...

Reply:
Relating what we spoke about offline, we decided to keep using the `user:hydra:doubleSided` attribute as it draws attention to the somewhat unusual nature of how `doubleSided` is handled with the `PxrSurface` shader. I'm shuffling `doubleSided` into that parameter in https://github.com/GafferHQ/gaffer/pull/6302/commits/474584a608bdf68e853391bc7ef0f73ffd3b78d0, as well as setting the `isDoubleSided` default to `1` to match Gaffer's default.

Reply:
We ended up deciding to _not_ shuffle into the `user:hydra:doubleSided` parameter and instead only default the double sided variable in `__UsdPreviewSurfaceParameters" to `1`. The `UsdPreviewSurfaceParameters` shader does not do true double sided - it renders the back side black when double sided is off, rather than transparent.

Since you can't get correct non-double sided behavior and it isn't clear which non-standard behavior to go with, we are opting for the less complex option of always using double sided USDPreviewSurface.

This is done in https://github.com/GafferHQ/gaffer/pull/6302/commits/f664dd94ba42cba3088c4286257e9566a3c85d5d